lib/stateTransition/StateTransitionFacade.js

```diff
@@ -19,37 +19,112 @@ const validateStateTransitionDataFactory = require('./validation/validateStateTr
 const validateDocumentsSTStructureFactory = require('../document/stateTransition/validation/structure/validateDocumentsSTStructureFactory');
 const validateDocumentFactory = require('../document/validateDocumentFactory');
 
+const fetchAndValidateDataContractFactory = require('../document/fetchAndValidateDataContractFactory');
 const enrichDataContractWithBaseDocument = require('../document/enrichDataContractWithBaseDocument');
 const findDuplicateDocumentsById = require('../document/stateTransition/validation/structure/findDuplicateDocumentsById');
 const findDuplicateDocumentsByIndices = require('../document/stateTransition/validation/structure/findDuplicateDocumentsByIndices');
 
+const validateDocumentsSTDataFactory = require('../document/stateTransition/validation/data/validateDocumentsSTDataFactory');
 const validateBlockchainUserFactory = require('./validation/validateBlockchainUserFactory');
+const fetchDocumentsFactory = require('../document/stateTransition/validation/data/fetchDocumentsFactory');
+const validateDocumentsUniquenessByIndicesFactory = require('../document/stateTransition/validation/data/validateDocumentsUniquenessByIndicesFactory');
+const getDataTriggersFactory = require('../dataTrigger/getDataTriggersFactory');
+const executeDataTriggersFactory = require('../document/stateTransition/validation/data/executeDataTriggersFactory');
 
 class StateTransitionFacade {
   /**
-   * @param {DashPlatformProtocol} dpp
+   * @param {DataProvider} dataProvider
    * @param {JsonSchemaValidator} validator
    */
-  constructor(dpp, validator) {
-    this.dpp = dpp;
+  constructor(dataProvider, validator) {
+    this.dataProvider = dataProvider;
     this.validator = validator;
 
     const validateDataContract = validateDataContractFactory(
-      this.validator,
+      validator,
     );
 
-    this.validateDataContractSTStructure = validateDataContractSTStructureFactory(
+    const validateDataContractSTStructure = validateDataContractSTStructureFactory(
       validateDataContract,
     );
 
-    this.validateDocument = validateDocumentFactory(
+    const validateDocument = validateDocumentFactory(
       this.validator,
       enrichDataContractWithBaseDocument,
     );
 
     this.createStateTransition = createStateTransitionFactory(
       createDataContract,
     );
+
+    const fetchAndValidateDataContract = fetchAndValidateDataContractFactory(
+      dataProvider,
+    );
+
+    const validateDocumentsSTStructure = validateDocumentsSTStructureFactory(
+      validateDocument,
+      findDuplicateDocumentsById,
+      findDuplicateDocumentsByIndices,
+      fetchAndValidateDataContract,
+    );
+
+    const typeExtensions = {
+      [stateTransitionTypes.DATA_CONTRACT]: {
+        function: validateDataContractSTStructure,
```

Comment:
Can we use a different name, like "action" maybe? I feel using keywords for property names can be a bad idea

Reply:
Not sure that I understand what do you mean. It's just a way how we pass ST specific logic as a dependency into State Transition validation function. We don't use it anywhere outside validateStateTransitionStructure.